src/main/java/seedu/duke/cli/Command.java

```diff
@@ -0,0 +1,19 @@
+package seedu.duke.cli;
+
+import seedu.duke.tasks.Task;
+
+import java.util.List;
+
+/**
+ * A command that performs actions on a task list.
+ */
+public interface Command {
+    /**
+     * Execute the command on the task list.
+     *
+     * @param taskList The task list to operate on
+     * @return {@code true} if the command ran successfully; {@code false} otherwise
+     * @throws CommandException if an error occurred while running the command
+     */
+    boolean execute(List<Task> taskList) throws CommandException;
```

Comment:
Maybe the name for the method could be renamed to sound more like a boolean.

Reply:
Ah yeah, I guess this method can become a `void` method because any errors should be raised as `CommandException`s anyway.